frontend/src/metabase/public/lib/code-templates.ts

```diff
@@ -1,29 +1,48 @@
+import { match } from "ts-pattern";
+
 import { optionsToHashParams } from "./embed";
 import type {
   CodeSampleParameters,
   EmbeddingDisplayOptions,
+  EmbeddingHashOptions,
   EmbeddingParametersValues,
 } from "./types";
 
 export function getIframeQueryWithoutDefaults(
   displayOptions: EmbeddingDisplayOptions,
 ) {
+  const hashOptions = transformEmbeddingDisplayToHashOptions(displayOptions);
+
   return optionsToHashParams(
-    removeDefaultValueParameters(displayOptions, {
+    removeDefaultValueParameters(hashOptions, {
       theme: "light",
       hide_download_button: false,
       background: true,
     }),
   );
 }
+
+function transformEmbeddingDisplayToHashOptions(
```

Comment:
Ideally this would be context-sensitive (e.g. only allow `pdf` and `results` if resourceType is dashboard), but I found that requires updating lots of places, so I'm not doing that for now.